client/session.go

```diff
@@ -841,23 +843,33 @@ func (s *session) Fetch(
 	return iter, nil
 }
 
-func (s *session) FetchAll(
-	namespace string,
-	ids []string,
+func (s *session) FetchIDs(
+	namespace ts.ID,
+	ids ts.IDsIterator,
+	startInclusive, endExclusive time.Time,
+) (encoding.SeriesIterators, error) {
+	nsID := s.idPool.Clone(namespace)
+	tsIDs := s.idPool.CloneIDs(ids)
+	return s.fetchOwnedIDs(nsID, tsIDs, startInclusive, endExclusive)
+}
+
+func (s *session) fetchOwnedIDs(
+	namespace ts.ID,
+	ids ts.IDs,
 	startInclusive, endExclusive time.Time,
 ) (encoding.SeriesIterators, error) {
 	f := s.fetchAttemptPool.Get()
-	f.args.namespace, f.args.ids, f.args.start, f.args.end =
-		namespace, ids, startInclusive, endExclusive
+	f.args.namespace, f.args.ids = namespace, ids
```

Comment:
Same problem as the write attempt with here, the first attempt will finalize these IDs and a retry would cause use after free.

Reply:
Addressed this one the same way as above.